File: app/src/test/java/com/boogie/vibestation/AppConfigTest.java
package com.boogie.vibestation;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

/**
 * Unit contract tests verifying AppConfig version formatting and presence.
 */
public class AppConfigTest {

    /**
     * Verifies that the APP_VERSION constant follows standard semver format.
     */
    @Test
    public void testAppVersionFormat() {
        assertNotNull(AppConfig.APP_VERSION);
        assertTrue("APP_VERSION should follow semantic versioning",
                AppConfig.APP_VERSION.matches("^\\d+\\.\\d+\\.\\d+$"));
        assertEquals("2.5.3", AppConfig.APP_VERSION);
    }
}

File: app/src/test/kotlin/com/boogie/vibestation/AppConfigTest.kt
package com.boogie.vibestation

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

/**
 * Unit contract tests verifying AppConfig version formatting and presence.
 */
class AppConfigTest {

    /**
     * Verifies that the APP_VERSION constant follows standard semver format.
     */
    @Test
    fun appVersionFormat() {
        assertTrue(
            AppConfig.APP_VERSION.matches(Regex("""^\d+\.\d+\.\d+$""")),
            "APP_VERSION should follow semantic versioning"
        )
        assertEquals("3.0.0", AppConfig.APP_VERSION)
    }
}
